pkg/infra/tower/tower-functions.go

```diff
@@ -0,0 +1,96 @@
+package tower
+
+import (
+	"log"
+	"math"
+	"math/rand"
+)
+
+func (t *Tower) initAgents() {
+	t.agents = make(map[string]BaseAgentCore, t.agentCount)
+}
+
+func (t *Tower) killAgent(id string) { // this removes the agent from the list of agents in the tower
+	log.Printf("Killing agent %s", id)
+	t.mx.RLock()
+	deadAgentFloor := t.agents[id].floor
+	agentType := t.agents[id].agentType
+	t.mx.RUnlock()
+	t.missingAgents[deadAgentFloor] = append(t.missingAgents[deadAgentFloor], agentType)
+	t.mx.Lock()
+	delete(t.agents, id)
+	t.mx.Unlock()
+}
```

Comment:
Why do you need a list of `missingAgents`? Surely an easier way of doing this is just in `simulation.go` `simulationLoop()` you check for which agents have `hp <= 0` and replace those agents??? That way you don't need to keep track of `missingAgents`? This just seems to me like you've got a lot of extra state you have to maintain and keep consistent?

Reply:
This would change how `hpDecay()` is written slightly... but I'm also not convinced `hpDecay()` should be concerned about whether an agent dies or not - it should really just be responsible for decaying HP and nothing else.

Reply:
1- Agreed. Future version we shall move states to simulation
2- Eventually `hpDecay()` would be parametrized, we'd like agent teams to PR proposals for it. We check in the function for death directly as otherwise we'd be checking right after it in a different loop without any agent state. Check can be moved to a different loop without issues